src/main/java/org/opensearch/knn/index/util/Nmslib.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.util;
+
+import com.google.common.collect.ImmutableMap;
+import org.opensearch.knn.index.KNNMethod;
+import org.opensearch.knn.index.KNNSettings;
+import org.opensearch.knn.index.MethodComponent;
+import org.opensearch.knn.index.Parameter;
+import org.opensearch.knn.index.SpaceType;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.opensearch.knn.common.KNNConstants.*;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION;
+
+/**
+ * Implements NativeLibrary for the nmslib native library
+ */
+class Nmslib extends NativeLibrary {
+
+    final static String HNSW_LIB_NAME = "hnsw";
+    final static String EXTENSION = ".hnsw";
```

Comment:
So this is a little bit confusing. HNSW_LIB_NAME is redundant with METHOD_HNSW, so I will delete that. 

EXTENSION is actually the libraries extension. Given that we are in the Nmslib class, I think it can be kept as EXTENSION. Ill add a comment.